server/sonar-db-dao/src/main/java/org/sonar/db/component/ComponentDao.java

```diff
@@ -104,19 +104,31 @@ public ComponentDto selectOrFailByUuid(DbSession session, String uuid) {
     return componentDto.get();
   }
 
+  /**
+   * Branches are filtered out. 
+   */
   public List<ComponentDto> selectByQuery(DbSession session, ComponentQuery query, int offset, int limit) {
     return selectByQueryImpl(session, null, query, offset, limit);
   }
 
+  /**
+   * Branches are filtered out. 
```

Comment:
No need for such javadoc, branches should only be returned by selectByKeyAndBranch and selectByKeysAndBranch